diff -ruN clementine-player-Clementine-d87307f.orig/3rdparty/libprojectm/CMakeLists.txt clementine-player-Clementine-d87307f/3rdparty/libprojectm/CMakeLists.txt
--- clementine-player-Clementine-d87307f.orig/3rdparty/libprojectm/CMakeLists.txt	2018-12-01 14:05:24.000000000 +0100
+++ clementine-player-Clementine-d87307f/3rdparty/libprojectm/CMakeLists.txt	2018-12-02 14:17:51.945163985 +0100
@@ -178,6 +178,8 @@
         COPY_ONLY
       )
     endforeach (preset)
+  elseif (WIN32)
+    INSTALL(FILES ${presets} DESTINATION ${CMAKE_INSTALL_PREFIX}/bin/projectm-presets)
   else (APPLE)
     INSTALL(FILES ${presets} DESTINATION ${CMAKE_INSTALL_PREFIX}/share/clementine/projectm-presets)
   endif (APPLE)
diff -ruN clementine-player-Clementine-d87307f.orig/3rdparty/libprojectm/glew.h clementine-player-Clementine-d87307f/3rdparty/libprojectm/glew.h
--- clementine-player-Clementine-d87307f.orig/3rdparty/libprojectm/glew.h	2018-12-01 14:05:24.000000000 +0100
+++ clementine-player-Clementine-d87307f/3rdparty/libprojectm/glew.h	2018-12-02 14:16:16.777667597 +0100
@@ -149,7 +149,7 @@
 #  endif
 #endif
 #if !defined(_PTRDIFF_T_DEFINED) && !defined(_PTRDIFF_T_)
-#  ifdef _WIN64
+#  if defined(_WIN64) && !defined(__MINGW32__)
 typedef __int64 ptrdiff_t;
 #  else
 typedef _W64 int ptrdiff_t;
