diff -ruN clementine-player-Clementine-95187ed.orig/src/musicbrainz/chromaprinter.cpp clementine-player-Clementine-95187ed/src/musicbrainz/chromaprinter.cpp
--- clementine-player-Clementine-95187ed.orig/src/musicbrainz/chromaprinter.cpp	2018-12-02 10:51:19.000000000 +0100
+++ clementine-player-Clementine-95187ed/src/musicbrainz/chromaprinter.cpp	2018-12-02 16:07:25.994394042 +0100
@@ -29,6 +29,10 @@
 #include "core/logging.h"
 #include "core/signalchecker.h"
 
+#ifndef u_int32_t
+typedef unsigned int u_int32_t;
+#endif
+
 static const int kDecodeRate = 11025;
 static const int kDecodeChannels = 1;
 static const int kPlayLengthSecs = 30;
