--- a/biosig.h	2015-03-06 10:56:16.000000000 +0100
+++ b/biosig.h	2015-03-21 20:30:26.371827905 +0100
@@ -40,17 +40,21 @@
 
 #define BIOSIG_VERSION_MAJOR 1
 #define BIOSIG_VERSION_MINOR 6
-#define BIOSIG_PATCHLEVEL 4
+#define BIOSIG_PATCHLEVEL 4
 // for backward compatibility 
 #define BIOSIG_VERSION_STEPPING BIOSIG_PATCHLEVEL	
 #define BIOSIG_VERSION (BIOSIG_VERSION_MAJOR * 10000 + BIOSIG_VERSION_MINOR * 100 + BIOSIG_PATCHLEVEL)
 // biosigCHECK_VERSION returns true if BIOSIG_VERSION is at least a.b.c
 #define biosigCHECK_VERSION(a,b,c) (BIOSIG_VERSION >= ( 10000*(a) + 100*(b) + (c) ) )
 
-#if defined(_VCPP_DEF) || defined(_MSC_VER)
-    #define __BYTE_ORDER  __LITTLE_ENDIAN
-    typedef long   		ssize_t;
-    typedef unsigned long	size_t;
+#if defined(_MSC_VER) && (_MSC_VER < 1600)
+#if defined(_WIN64)
+    typedef __int64		ssize_t;
+    typedef unsigned __int64	size_t;
+#else
+    typedef __int32		ssize_t;
+    typedef unsigned __int32	size_t;
+#endif
     typedef unsigned __int64	uint64_t;
     typedef __int64		int64_t;
     typedef unsigned __int32	uint32_t;
