diff --git a/biosig4c++/Makefile.in b/biosig4c++/Makefile.in
index e692f579..b943a011 100644
--- a/biosig4c++/Makefile.in
+++ b/biosig4c++/Makefile.in
@@ -727,8 +727,8 @@ ifeq (,$(findstring Darwin, $(shell uname)))   # not on MacOSX
 	ln -sf "$<" "$@"
 endif
 
-ifneq ($(TARGETPLATFORM),windows)
 ifeq (@SHARED_LIBS@,yes)
+ifneq ($(TARGETPLATFORM),windows)
 $(DLPRE)biosig${FULLDLEXT}:$(OBJECTS)
 	$(LD) $(LDFLAGS) ${SONAME_PREFIX}"$@" $^ $(SHAREDLIB) $(LDLIBS) -o "$@"
 
@@ -740,7 +740,6 @@ $(DLPRE)gdftime${FULLDLEXT}:gdftime.o
 
 $(DLPRE)physicalunits${FULLDLEXT}:physicalunits.o
 	$(LD) $(LDFLAGS) ${SONAME_PREFIX}"$@" $^ $(SHAREDLIB) $(LDLIBS) -o "$@"
-endif
 else
 #############################################################
 #	Windows DLL (mingw, cygwin
@@ -759,6 +758,7 @@ $(DLPRE)biosig${FULLDLEXT}: $(OBJECTS) libbiosig.pc
 	$(LD) $(LDFLAGS) -o "$@" -s -shared -fPIC $(OBJECTS) $(LDLIBS) -Wl,-subsystem,windows,--output-def,libbiosig.def,--out-implib,libbiosig.dll.a
 
 endif
+endif
 
 #############################################################
 #	Static Libaries
@@ -802,17 +802,17 @@ libbiosig.pc :
 	echo "Libs.private: $(LDLIBS)"      >> "$@"
 
 ## save2gdf, pdp2gdf
-%${BINEXT}: %.c $(DLPRE)biosig$(FULLDLEXT)
+%${BINEXT}: %.c libbiosig
 	$(CC) $(DEFINES) $(CPPFLAGS) $(CFLAGS) $(LDFLAGS) "$<" -L. -lbiosig -lstdc++ $(LDLIBS) -o "$@"
 
 physicalunits${BINEXT} : pu.c physicalunits.o
 	$(CC) $(DEFINES) $(CPPFLAGS) $(CFLAGS) $(LDFLAGS) $^ $(LDLIBS) -o "$@"
 
 ifeq (1,@HAVE_LIBB64@)
-biosig_fhir${BINEXT}: biosig_fhir.c $(DLPRE)biosig$(FULLDLEXT)
+biosig_fhir${BINEXT}: biosig_fhir.c libbiosig
 	$(CC) $(DEFINES) $(CPPFLAGS) $(CFLAGS) $(LDFLAGS) "$<" -L. -lbiosig -lstdc++ -lb64 $(LDLIBS) -o "$@"
 else
-biosig_fhir${BINEXT}: biosig_fhir.c libb64-1.2.1/src/cencode.c $(DLPRE)biosig$(FULLDLEXT)
+biosig_fhir${BINEXT}: biosig_fhir.c libb64-1.2.1/src/cencode.c libbiosig
 	$(CC) $(DEFINES) $(CPPFLAGS) $(CFLAGS) $(LDFLAGS) -Ilibb64-1.2.1/include/ "$<" libb64-1.2.1/src/cencode.c -L. -lbiosig -lstdc++ $(LDLIBS) -o "$@"
 endif
 
