diff --git a/biosig4c++/Makefile b/biosig4c++/Makefile
index a2bc2a4..2770ca2 100644
--- a/Makefile
+++ b/Makefile
@@ -635,18 +635,18 @@ t240/libcnsfef.a:
 	$(MAKE) -C t240 libcnsfef.a CC=$(CC) AR='$(AR)'	# generate t240/*.o files
 
 libbiosig.${FULLDLEXT}: $(OBJECTS) biosig.o t240/libcnsfef.a
-	$(LD) $(LDFLAGS) $(LIBS) ${SONAME_PREFIX}libbiosig${SONAME_POSTFIX} $(OBJECTS) t240/*.o -o "$@"
+	$(LD) ${SONAME_PREFIX}libbiosig${SONAME_POSTFIX} $(OBJECTS) t240/*.o $(LDFLAGS) $(LIBS) -o "$@"
 
 libbiosig2.${FULLDLEXT}: $(OBJECTS)  biosig2.o t240/libcnsfef.a
-	$(LD) $(LDFLAGS) $(LIBS) ${SONAME_PREFIX}libbiosig2${SONAME_POSTFIX} $(OBJECTS) biosig2.o t240/*.o -o "$@"
+	$(LD) ${SONAME_PREFIX}libbiosig2${SONAME_POSTFIX} $(OBJECTS) biosig2.o t240/*.o $(LDFLAGS) $(LIBS) -o "$@"
 
 libgdf.${FULLDLEXT}: gdf.o getlogin.o gdftime.o physicalunits.o
-	$(LD) $(LDFLAGS) $(LIBS) ${SONAME_PREFIX}libgdf${SONAME_POSTFIX} $? -o "$@"
+	$(LD) ${SONAME_PREFIX}libgdf${SONAME_POSTFIX} $? $(LDFLAGS) $(LIBS) -o "$@"
 
 libgdftime.${FULLDLEXT}: gdftime.o
-	$(LD) $(LDFLAGS) $(LIBS) ${SONAME_PREFIX}libgdftime${SONAME_POSTFIX} $? -o "$@"
+	$(LD) ${SONAME_PREFIX}libgdftime${SONAME_POSTFIX} $? $(LDFLAGS) $(LIBS) -o "$@"
 libphysicalunits.${FULLDLEXT}: physicalunits.o
-	$(LD) $(LDFLAGS) $(LIBS) ${SONAME_PREFIX}libgdftime${SONAME_POSTFIX} $? -o "$@"
+	$(LD) ${SONAME_PREFIX}libgdftime${SONAME_POSTFIX} $? $(LDFLAGS) $(LIBS) -o "$@"
 
 %.${DLEXT}: %.${FULLDLEXT}
 	ln -sf "$(*F).${FULLDLEXT}"	"$(*F)${SONAME_POSTFIX}"
diff --git a/biosig4c++/t210/sopen_heka_read.c b/biosig4c++/t210/sopen_heka_read.c
index 61e8da7..b4d6603 100644
--- a/t210/sopen_heka_read.c
+++ b/t210/sopen_heka_read.c
@@ -577,7 +577,7 @@ if (VERBOSE_LEVEL>6) fprintf(stdout,"HEKA L5 @%i= #%i,%i, %s %g/%g %g/%g \n",(in
                                                         biosigERROR(hdr, B4C_FORMAT_UNSUPPORTED, "Heka/Patchmaster: Yoffset is not zero");
 						}
 						if (hdr->AS.Header[pos+220] != 1) {
-                                                        biosigERROR(hdr, B4C_FORMAT_UNSUPPORTED, "Heka/Patchmaster: ValidYRange not set");
+//                                                        biosigERROR(hdr, B4C_FORMAT_UNSUPPORTED, "Heka/Patchmaster: ValidYRange not set");
 						}
 
 if (VERBOSE_LEVEL>7) fprintf(stdout,"HEKA L6 @%i= #%i,%i, %s %f-%fHz\t%i/%i %i/%i %i/%i %i/%i \n",(int)(pos+StartOfData),ns,AdcChan,Label,hdr->SampleRate,Fs,k1,K1,k2,K2,k3,K3,k4,K4);
