diff --git a/t210/sopen_cfs_read.c b/t210/sopen_cfs_read.c
index 9fe1871..873a056 100644
--- a/t210/sopen_cfs_read.c
+++ b/t210/sopen_cfs_read.c
@@ -81,9 +81,9 @@ EXTERN_C void sopen_cfs_read(HDRTYPE* hdr) {
 			t.tm_min  = atoi(strtok(NULL,":/"));
 			t.tm_sec  = atoi(strtok(NULL,":/"));
 			memcpy(strtmp, hdr->AS.Header+0x22, 8);
-			t.tm_mday = atoi(strtok(strtmp,":/"));
-			t.tm_mon  = atoi(strtok(NULL,":/"))-1;
-			t.tm_year = atoi(strtok(NULL,":/"));
+			t.tm_mday = atoi(strtok(strtmp,"-:/"));
+			t.tm_mon  = atoi(strtok(NULL,"-:/"))-1;
+			t.tm_year = atoi(strtok(NULL,"-:/"));
 			if (t.tm_year<39) t.tm_year+=100;
 			hdr->T0 = tm_time2gdf_time(&t);
 		}
