diff -uNr cairo-1.10.2-orig/src/cairo-system.c cairo-1.10.2/src/cairo-system.c
--- cairo-1.10.2-orig/src/cairo-system.c	2010-06-18 13:47:14.000000000 +0200
+++ cairo-1.10.2/src/cairo-system.c	2012-04-14 22:01:27.848928137 +0200
@@ -70,12 +70,12 @@
 
 /* declare to avoid "no previous prototype for 'DllMain'" warning */
 BOOL WINAPI
-DllMain (HINSTANCE hinstDLL,
+cairo_DllMain (HINSTANCE hinstDLL,
          DWORD     fdwReason,
          LPVOID    lpvReserved);
 
 BOOL WINAPI
-DllMain (HINSTANCE hinstDLL,
+cairo_DllMain (HINSTANCE hinstDLL,
          DWORD     fdwReason,
          LPVOID    lpvReserved)
 {
