--- cmake/FindGLEW.cmake.orig	2016-07-19 14:11:06.255586305 +0200
+++ cmake/FindGLEW.cmake	2016-07-19 14:11:35.454653035 +0200
@@ -4,12 +4,12 @@
 include(FindPackageHandleStandardArgs)
 
 find_path(GLEW_H_PATH NAMES GL/glew.h)
-find_library(GLEW_LIB NAMES GLEW libGLEW glew libglew PATH_SUFFIXES dynamic)
+find_library(GLEW_LIB NAMES glew32 GLEW libGLEW glew libglew PATH_SUFFIXES dynamic)
 find_library(GLEW_LIB_DBG NAMES GLEW_d libGLEW_d glew_d libglew_d PATH_SUFFIXES dynamic)
 mark_as_advanced(GLEW_H_PATH GLEW_LIB GLEW_LIB_DBG)
 
 if (WIN32 OR APPLE)
-    find_library(GLEW_LIB_STATIC NAMES GLEW libGLEW glew libglew PATH_SUFFIXES static)
+    find_library(GLEW_LIB_STATIC NAMES glew32 GLEW libGLEW glew libglew PATH_SUFFIXES static)
     find_library(GLEW_LIB_STATIC_DBG NAMES GLEW_d libGLEW_d glew_d libglew_d PATH_SUFFIXES static)
     mark_as_advanced(GLEW_LIB_STATIC GLEW_LIB_STATIC_DBG)
 endif()
