diff --git a/include/chipmunk/chipmunk.h b/include/chipmunk/chipmunk.h
index 616bfe0..5fbb390 100644
--- a/include/chipmunk/chipmunk.h
+++ b/include/chipmunk/chipmunk.h
@@ -25,7 +25,7 @@
 #include <stdlib.h>
 #include <math.h>
 
-#ifdef WIN32
+#if defined(WIN32) || defined(__WIN32) || defined(__WIN32__)
 	// For alloca().
 	#include <malloc.h>
 	#define CP_EXPORT __declspec(dllexport)
diff --git a/src/cpHastySpace.c b/src/cpHastySpace.c
index 4e1bf2c..a3ad440 100644
--- a/src/cpHastySpace.c
+++ b/src/cpHastySpace.c
@@ -6,7 +6,9 @@
 
 #include <pthread.h>
 //#include <sys/param.h >
+#ifndef _WIN32
 #include <sys/sysctl.h>
+#endif
 
 #include "chipmunk/chipmunk_private.h"
 #include "chipmunk/cpHastySpace.h"
