--- Makefile.in	2015-06-18 16:04:54.000000000 -0400
+++ Makefile_patch.in	2018-08-19 11:01:39.044928227 -0400
@@ -1150,10 +1150,9 @@
 	    ($(MKDIR) -p $(DESTDIR)$(bindir) && \
 	    $(CHMOD) $(dmode) $(DESTDIR)$(bindir))
 	@for i in $(UTIL_PROGS); do \
-		test -f $$i.exe && i=$$i.exe; \
-		e=`echo $$i | $(SED) '$(transform)'`; \
+		e=$$i.exe; \
 		$(RM) $(DESTDIR)$(bindir)/$$e; \
-		$(INSTALLER) $$i $(DESTDIR)$(bindir)/$$e; \
+		$(INSTALLER) $$e $(DESTDIR)$(bindir)/$$e; \
 		$(STRIP) $(DESTDIR)$(bindir)/$$e; \
 		$(CHMOD) $(emode) $(DESTDIR)$(bindir)/$$e; \
 	done
