diff -bur gdal-1.11.1.orig/configure.in gdal-1.11.1/configure.in
--- gdal-1.11.1.orig/configure.in	2014-10-24 10:57:41.753494871 +0530
+++ gdal-1.11.1/configure.in	2014-10-24 10:58:29.813123435 +0530
@@ -1935,14 +1935,26 @@
     if test "$HAVE_OPENJPEG" = "yes"; then
         LIBS="-lopenjp2 $LIBS"
     fi
+  else
+    AC_CHECK_HEADERS([openjpeg-2.1/openjpeg.h])
+    if test "$ac_cv_header_openjpeg_2_1_openjpeg_h" = "yes"; then
+        AC_CHECK_LIB(openjp2,opj_stream_set_user_data_length,HAVE_OPENJPEG=yes,HAVE_OPENJPEG=no,)
+        if test "$HAVE_OPENJPEG" = "yes"; then
+            OPENJPEG_VERSION=20100
+            LIBS="-lopenjp2 $LIBS"
+        fi
+    fi
   fi
 else
 
   HAVE_OPENJPEG=yes
   if test -r $with_openjpeg/include/openjpeg-2.0/openjpeg.h ; then
     EXTRA_INCLUDES="-I$with_openjpeg/include $EXTRA_INCLUDES"
+  elif test -r $with_openjpeg/include/openjpeg-2.1/openjpeg.h ; then
+    OPENJPEG_VERSION=20100
+    EXTRA_INCLUDES="-I$with_openjpeg/include $EXTRA_INCLUDES"
   else
-    AC_MSG_ERROR([openjpeg.h not found in $with_openjpeg/include/openjpeg-2.0])
+    AC_MSG_ERROR([openjpeg.h not found in $with_openjpeg/include/openjpeg-2.0 or $with_openjpeg/include/openjpeg-2.1])
   fi
 
   AC_CHECK_LIB(openjp2,opj_stream_set_user_data_length,HAVE_OPENJPEG=yes,HAVE_OPENJPEG=no,-L$with_openjpeg/lib)
@@ -1958,7 +1970,7 @@
 fi
 
 AC_SUBST([HAVE_OPENJPEG],$HAVE_OPENJPEG)
-
+AC_SUBST([OPENJPEG_VERSION],$OPENJPEG_VERSION)
 
 dnl ---------------------------------------------------------------------------
 dnl Select a FGDB API to use, or disable driver.
diff -bur gdal-1.11.1.orig/frmts/openjpeg/GNUmakefile gdal-1.11.1/frmts/openjpeg/GNUmakefile
--- gdal-1.11.1.orig/frmts/openjpeg/GNUmakefile	2014-10-24 10:57:41.659493642 +0530
+++ gdal-1.11.1/frmts/openjpeg/GNUmakefile	2014-10-24 10:58:54.512446432 +0530
@@ -5,6 +5,10 @@
 
 CPPFLAGS	:=	$(GDAL_INCLUDE) $(CPPFLAGS)
 
+ifneq ($(OPENJPEG_VERSION),)
+CPPFLAGS 	:=	$(CPPFLAGS) -DOPENJPEG_VERSION=$(OPENJPEG_VERSION)
+endif
+
 default:	$(OBJ:.o=.$(OBJ_EXT))
 
 clean:
diff -bur gdal-1.11.1.orig/frmts/openjpeg/makefile.vc gdal-1.11.1/frmts/openjpeg/makefile.vc
--- gdal-1.11.1.orig/frmts/openjpeg/makefile.vc	2014-10-24 10:57:41.659493642 +0530
+++ gdal-1.11.1/frmts/openjpeg/makefile.vc	2014-10-24 10:58:45.758332006 +0530
@@ -5,7 +5,11 @@
 
 !INCLUDE $(GDAL_ROOT)\nmake.opt
 
-EXTRAFLAGS = $(OPENJPEG_CFLAGS)
+EXTRAFLAGS = $(OPENJPEG_CFLAGS) $(OPENJPEG_VERSION_CFLAGS)
+
+!IFDEF OPENJPEG_VERSION
+OPENJPEG_VERSION_CFLAGS = -DOPENJPEG_VERSION=$(OPENJPEG_VERSION)
+!ENDIF
 
 default:	$(OBJ)
 	xcopy /D  /Y *.obj ..\o
diff -bur gdal-1.11.1.orig/frmts/openjpeg/openjpegdataset.cpp gdal-1.11.1/frmts/openjpeg/openjpegdataset.cpp
--- gdal-1.11.1.orig/frmts/openjpeg/openjpegdataset.cpp	2014-10-24 10:57:41.659493642 +0530
+++ gdal-1.11.1/frmts/openjpeg/openjpegdataset.cpp	2014-10-24 10:59:03.411562836 +0530
@@ -29,8 +29,12 @@
 
 /* This file is to be used with openjpeg 2.0 */
 
+#if defined(OPENJPEG_VERSION) && OPENJPEG_VERSION >= 20100
+#include <openjpeg-2.1/openjpeg.h>
+#else
 #include <stdio.h> /* openjpeg.h needs FILE* */
 #include <openjpeg-2.0/openjpeg.h>
+#endif
 #include <vector>
 
 #include "gdaljp2abstractdataset.h"
@@ -612,7 +616,11 @@
     opj_stream_set_read_function(pStream, JP2OpenJPEGDataset_Read);
     opj_stream_set_seek_function(pStream, JP2OpenJPEGDataset_Seek);
     opj_stream_set_skip_function(pStream, JP2OpenJPEGDataset_Skip);
+#if defined(OPENJPEG_VERSION) && OPENJPEG_VERSION >= 20100
+    opj_stream_set_user_data(pStream, fp, NULL);
+#else
     opj_stream_set_user_data(pStream, fp);
+#endif
 
     if(!opj_read_header(pStream,pCodec,&psImage))
     {
@@ -950,7 +958,11 @@
     opj_stream_set_read_function(pStream, JP2OpenJPEGDataset_Read);
     opj_stream_set_seek_function(pStream, JP2OpenJPEGDataset_Seek);
     opj_stream_set_skip_function(pStream, JP2OpenJPEGDataset_Skip);
+#if defined(OPENJPEG_VERSION) && OPENJPEG_VERSION >= 20100
+    opj_stream_set_user_data(pStream, fp, NULL);
+#else
     opj_stream_set_user_data(pStream, fp);
+#endif
 
     opj_image_t * psImage = NULL;
     OPJ_INT32  nX0,nY0;
@@ -1498,7 +1510,11 @@
     opj_stream_set_write_function(pStream, JP2OpenJPEGDataset_Write);
     opj_stream_set_seek_function(pStream, JP2OpenJPEGDataset_Seek);
     opj_stream_set_skip_function(pStream, JP2OpenJPEGDataset_Skip);
+#if defined(OPENJPEG_VERSION) && OPENJPEG_VERSION >= 20100
+    opj_stream_set_user_data(pStream, fp, NULL);
+#else
     opj_stream_set_user_data(pStream, fp);
+#endif
 
     if (!opj_start_compress(pCodec,psImage,pStream))
     {
Only in gdal-1.11.1/frmts/openjpeg: openjpegdataset.cpp.orig
diff -bur gdal-1.11.1.orig/GDALmake.opt.in gdal-1.11.1/GDALmake.opt.in
--- gdal-1.11.1.orig/GDALmake.opt.in	2014-10-24 10:57:41.482491327 +0530
+++ gdal-1.11.1/GDALmake.opt.in	2014-10-24 10:59:09.624644048 +0530
@@ -191,6 +191,7 @@
 # JPEG-2000 Support via OpenJPEG library.
 #
 HAVE_OPENJPEG   = @HAVE_OPENJPEG@
+OPENJPEG_VERSION = @OPENJPEG_VERSION@
 
 #
 # MrSID support via LizardTech's DSDK
diff -bur gdal-1.11.1.orig/nmake.opt gdal-1.11.1/nmake.opt
--- gdal-1.11.1.orig/nmake.opt	2014-10-24 10:57:41.753494871 +0530
+++ gdal-1.11.1/nmake.opt	2014-10-24 10:58:22.723030729 +0530
@@ -522,7 +522,9 @@
 # Uncomment for OpenJpeg (release v2.0.0) support
 #OPENJPEG_ENABLED = YES
 #OPENJPEG_CFLAGS = -IC:\openjpeg\include
-#OPENJPEG_LIB = C:\openjpeg\lib\openjpeg.lib
+#OPENJPEG_LIB = C:\openjpeg\lib\openjp2.lib
+# For OpenJpeg >= 2.1, uncomment
+#OPENJPEG_VERSION = 20100
 
 # Uncomment for PDF support
 # Uncomment POPPLER_BASE_STREAM_HAS_TWO_ARGS = YES for Poppler >= 0.16.0
