diff --git a/gdk-pixbuf-2.0.pc.in b/gdk-pixbuf-2.0.pc.in
index 7addefc..a07a46f 100644
--- a/gdk-pixbuf-2.0.pc.in
+++ b/gdk-pixbuf-2.0.pc.in
@@ -12,7 +12,7 @@ Name: GdkPixbuf
 Description: Image loading and scaling
 Version: @VERSION@
 Requires: gobject-2.0
-Requires.private: gmodule-no-export-2.0 @PNG_DEP_CFLAGS_PACKAGES@
+Requires.private: gmodule-no-export-2.0 @PNG_DEP_CFLAGS_PACKAGES@ libtiff-4
 Libs: -L${libdir} -lgdk_pixbuf-@GDK_PIXBUF_API_VERSION@
 Libs.private: @GDK_PIXBUF_EXTRA_LIBS@
 Cflags: -I${includedir}/gdk-pixbuf-@GDK_PIXBUF_API_VERSION@ @GDK_PIXBUF_EXTRA_CFLAGS@
