diff --git a/configure.ac b/configure.ac
index d929c84..20f8121 100644
--- a/configure.ac
+++ b/configure.ac
@@ -126,16 +126,16 @@ case $host in
     ;;
 esac
 
-if test "$os_win32" = "yes"; then
-  if test x$enable_static = xyes -o x$enable_static = x; then
-    AC_MSG_WARN([Disabling static library build, must build as DLL on Windows.])
-    enable_static=no
-  fi
-  if test x$enable_shared = xno; then
-    AC_MSG_WARN([Enabling shared library build, must build as DLL on Windows.])
-  fi
-  enable_shared=yes
-fi
+dnl if test "$os_win32" = "yes"; then
+dnl   if test x$enable_static = xyes -o x$enable_static = x; then
+dnl     AC_MSG_WARN([Disabling static library build, must build as DLL on Windows.])
+dnl     enable_static=no
+dnl   fi
+dnl   if test x$enable_shared = xno; then
+dnl     AC_MSG_WARN([Enabling shared library build, must build as DLL on Windows.])
+dnl   fi
+dnl   enable_shared=yes
+dnl fi
 
 dnl Initialize libtool
 LT_PREREQ([2.2.6])
