diff -bur InsightToolkit-3d95bcf/Modules/IO/TIFF/src/itkTIFFImageIO.cxx InsightToolkit-3d95bcf-new/Modules/IO/TIFF/src/itkTIFFImageIO.cxx
--- InsightToolkit-3d95bcf/Modules/IO/TIFF/src/itkTIFFImageIO.cxx	2014-07-21 08:41:32.779746744 +0200
+++ InsightToolkit-3d95bcf-new/Modules/IO/TIFF/src/itkTIFFImageIO.cxx	2014-07-21 09:28:01.717936089 +0200
@@ -20,7 +20,11 @@
 #include "itksys/SystemTools.hxx"
 
 #include <sys/stat.h>
-
+#ifdef __MSVCRT__
+#if __MSVCRT_VERSION__ < 0x0601
+#define _stat64 _stati64
+#endif
+#endif
 #include "itk_tiff.h"
 
 namespace itk
