diff -burN InsightToolkit-4.7.1.orig/Modules/ThirdParty/KWSys/src/KWSys/SharedForward.h.in InsightToolkit-4.7.1/Modules/ThirdParty/KWSys/src/KWSys/SharedForward.h.in
--- InsightToolkit-4.7.1.orig/Modules/ThirdParty/KWSys/src/KWSys/SharedForward.h.in	2015-04-28 13:08:51.635284316 +0200
+++ InsightToolkit-4.7.1/Modules/ThirdParty/KWSys/src/KWSys/SharedForward.h.in	2015-04-28 13:10:27.023288933 +0200
@@ -513,7 +513,7 @@
 #if defined(_MSC_VER)
   _execvp(cmd, argv);
 #elif defined(__MINGW32__) && !defined(__MINGW64__)
-  execvp(cmd, argv);
+  execvp(cmd, (char* const*)argv);
 #else
   execvp(cmd, (char* const*)argv);
 #endif
