diff -burN InsightToolkit-4.8.2.orig/Modules/ThirdParty/GDCM/src/gdcm/Utilities/socketxx/socket++/CMakeLists.txt InsightToolkit-4.8.2/Modules/ThirdParty/GDCM/src/gdcm/Utilities/socketxx/socket++/CMakeLists.txt
--- InsightToolkit-4.8.2.orig/Modules/ThirdParty/GDCM/src/gdcm/Utilities/socketxx/socket++/CMakeLists.txt	2015-12-02 13:19:53.030344390 +0100
+++ InsightToolkit-4.8.2/Modules/ThirdParty/GDCM/src/gdcm/Utilities/socketxx/socket++/CMakeLists.txt	2015-12-02 13:32:36.094384138 +0100
@@ -6,6 +6,8 @@
 include(CheckFunctionExists)
 CHECK_FUNCTION_EXISTS(strsignal SOCKETXX_HAVE_STRSIGNAL)
 
+add_definitions(-Dsocketxx_EXPORTS)
+
 # specific
 if(WIN32)
 else()
@@ -67,6 +69,8 @@
     )
 endif()
 
+set(GDCM_BUILD_SHARED_LIBS ${BUILD_SHARED_LIBS})
+
 add_library(${SOCKETXX_LIBRARY_NAME} ${libsocket___la_SOURCES})
 if(WIN32)
   target_link_libraries(${SOCKETXX_LIBRARY_NAME} ws2_32.lib)
@@ -103,4 +107,3 @@
     DESTINATION ${SOCKETXX_INSTALL_INCLUDE_DIR} COMPONENT Headers
   )
 endif()
-
