diff -burN ITK.original/Modules/ThirdParty/KWSys/src/KWSys/SharedForward.h.in ITK/Modules/ThirdParty/KWSys/src/KWSys/SharedForward.h.in
--- ITK.original/Modules/ThirdParty/KWSys/src/KWSys/SharedForward.h.in	2014-07-30 03:38:46.547938233 +0200
+++ ITK/Modules/ThirdParty/KWSys/src/KWSys/SharedForward.h.in	2014-07-30 03:37:16.114246728 +0200
@@ -512,6 +512,8 @@
   /* Invoke the child process.  */
 #if defined(_MSC_VER)
   _execvp(cmd, argv);
+#elif defined(__MINGW64__) || defined(__MINGW32__)
+  execvp(cmd, (char* const*)argv);
 #elif defined(__MINGW32__)
   execvp(cmd, argv);
 #else
