diff --git a/configure b/configure
index 0382145..da5e49d 100755
--- a/configure
+++ b/configure
@@ -23815,7 +23815,6 @@ fi
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: ${camlibdir}" >&5
 $as_echo "${camlibdir}" >&6; }
 
-AM_CPPFLAGS="$AM_CPPFLAGS -DCAMLIBS=\\\"\$(camlibdir)\\\""
 
 
 # Extract the first word of "cmp", so it can be a program name with args.
diff --git a/configure.ac b/configure.ac
index 4b5e12d..f89055e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -521,7 +534,6 @@ AC_ARG_WITH([camlibdir],[AS_HELP_STRING(
 ])
 AC_MSG_RESULT([${camlibdir}])
 AC_SUBST([camlibdir])
-AM_CPPFLAGS="$AM_CPPFLAGS -DCAMLIBS=\\\"\$(camlibdir)\\\""
 
 
 dnl ---------------------------------------------------------------------------
diff --git a/gphoto2/gphoto2-abilities-list.h b/gphoto2/gphoto2-abilities-list.h
index 58eb3d0..e92836f 100644
--- a/gphoto2/gphoto2-abilities-list.h
+++ b/gphoto2/gphoto2-abilities-list.h
@@ -196,6 +196,11 @@ const char *gp_message_codeset (const char *);
  */
 #ifdef _GPHOTO2_INTERNAL_CODE
 #define CAMLIBDIR_ENV "CAMLIBS"
+#ifdef WIN32
+#ifndef CAMLIBS
+#define CAMLIBS "./libgphoto2"
+#endif
+#endif
 #endif /* _GPHOTO2_INTERNAL_CODE */
 
 
diff --git a/gphoto2/gphoto2.h b/gphoto2/gphoto2.h
index 9bb6ba9..7654134 100644
--- a/gphoto2/gphoto2.h
+++ b/gphoto2/gphoto2.h
@@ -39,7 +39,7 @@ extern "C" {
 
 #ifdef WIN32
 #ifndef CAMLIBS
-#define CAMLIBS "."
+#define CAMLIBS "./libgphoto2"
 #endif
 #endif
 
diff --git a/libgphoto2_port/configure b/libgphoto2_port/configure
index 10ef797..9decad8 100755
--- a/libgphoto2_port/configure
+++ b/libgphoto2_port/configure
@@ -17493,7 +17493,6 @@ done
 
 iolibdir="\$(libdir)/\$(PACKAGE_TARNAME)/\$(VERSION)"
 
-AM_CPPFLAGS="$AM_CPPFLAGS -DIOLIBS=\\\"${iolibdir}\\\""
 
 sorted_iolib_list="$(echo "${IOLIB_LIST}" | tr ' ' '\n' | sort | ${SED} '/^$/d' | tr '\n' ' ' | ${SED} 's/ $//')"
 
diff --git a/libgphoto2_port/configure.ac b/libgphoto2_port/configure.ac
index c9ab924..3ba281b 100644
--- a/libgphoto2_port/configure.ac
+++ b/libgphoto2_port/configure.ac
@@ -468,7 +468,6 @@ for x in ${IOLIB_LIST}; do
 done
 AC_SUBST(IOLIB_LTLIST)
 AC_SUBST([iolibdir],["\$(libdir)/\$(PACKAGE_TARNAME)/\$(VERSION)"])
-AM_CPPFLAGS="$AM_CPPFLAGS -DIOLIBS=\\\"${iolibdir}\\\""
 
 sorted_iolib_list="$(echo "${IOLIB_LIST}" | tr ' ' '\n' | sort | ${SED} '/^$/d' | tr '\n' ' ' | ${SED} 's/ $//')"
 AC_DEFINE_UNQUOTED([IOLIB_LIST], ["${sorted_iolib_list}"], [Define as string containing a list of the iolibs])
diff --git a/libgphoto2_port/gphoto2/gphoto2-port-portability.h b/libgphoto2_port/gphoto2/gphoto2-port-portability.h
index 1a8cdc1..bdf8147 100644
--- a/libgphoto2_port/gphoto2/gphoto2-port-portability.h
+++ b/libgphoto2_port/gphoto2/gphoto2-port-portability.h
@@ -38,7 +38,7 @@
 # include <direct.h>
 
 # ifndef IOLIBS
-#  define IOLIBS			"."
+#  define IOLIBS		"./libgphoto2_port"
 # endif
 # define strcasecmp		_stricmp
 # ifndef snprintf
