diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6661f8b..73d098b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -636,14 +636,6 @@ if (LWS_WITH_STATIC)
 				${HDR_PUBLIC}
 				${SOURCES})
 	list(APPEND LWS_LIBRARIES websockets)
-
-	if (WIN32)
-		# Windows uses the same .lib ending for static libraries and shared
-		# library linker files, so rename the static library.
-		set_target_properties(websockets
-			PROPERTIES
-			OUTPUT_NAME websockets_static)
-	endif()
 endif()
 
 if (LWS_WITH_SHARED)
diff --git a/cmake/LibwebsocketsConfig.cmake.in b/cmake/LibwebsocketsConfig.cmake.in
index bea82b5..6973fbf 100644
--- a/cmake/LibwebsocketsConfig.cmake.in
+++ b/cmake/LibwebsocketsConfig.cmake.in
@@ -13,5 +13,5 @@ set(LIBWEBSOCKETS_INCLUDE_DIRS "@LWS__INCLUDE_DIRS@")
 include(${LWS_CMAKE_DIR}/LibwebsocketsTargets.cmake)
 
 # IMPORTED targets from LibwebsocketsTargets.cmake
-set(LIBWEBSOCKETS_LIBRARIES websockets websockets_shared)
+set(LIBWEBSOCKETS_LIBRARIES websockets)
 
