diff -burN opencv-2.4.9.orig/modules/highgui/src/window_w32.cpp opencv-2.4.9/modules/highgui/src/window_w32.cpp
--- opencv-2.4.9.orig/modules/highgui/src/window_w32.cpp	2014-10-08 17:11:38.841289897 +0530
+++ opencv-2.4.9/modules/highgui/src/window_w32.cpp	2014-10-08 17:12:09.875635237 +0530
@@ -60,7 +60,7 @@
 #include <vector>
 #include <functional>
 #include "opencv2/highgui/highgui.hpp"
-#include <GL\gl.h>
+#include <GL/gl.h>
 #endif
 
 static const char* trackbar_text =
