diff -burN OpenThreads-2.6.0.orig/CMakeLists.txt OpenThreads-2.6.0/CMakeLists.txt
--- OpenThreads-2.6.0.orig/CMakeLists.txt	2015-06-21 13:09:05.902865270 +0200
+++ OpenThreads-2.6.0/CMakeLists.txt	2015-06-21 13:09:31.246877234 +0200
@@ -67,7 +67,7 @@
 # Use our modified version of FindThreads.cmake which has Sproc hacks.
 FIND_PACKAGE(Threads)
 
-SUBDIRS(src/OpenThreads examples)
+SUBDIRS(src/OpenThreads)
 
 # FIXME: Make optional install test programs
 # FIXME: /bin directory not correct for Windows
diff -burN OpenThreads-2.6.0.orig/src/OpenThreads/CMakeLists.txt OpenThreads-2.6.0/src/OpenThreads/CMakeLists.txt
--- OpenThreads-2.6.0.orig/src/OpenThreads/CMakeLists.txt	2015-06-21 13:09:05.891865265 +0200
+++ OpenThreads-2.6.0/src/OpenThreads/CMakeLists.txt	2015-06-21 13:11:46.054870570 +0200
@@ -116,9 +116,9 @@
 
 # Generate pkg-config configuration file
 
-CONFIGURE_FILE(${PROJECT_SOURCE_DIR}/packaging/pkgconfig/openthreads.pc.in
-        ${PROJECT_BINARY_DIR}/packaging/pkgconfig/openthreads.pc
-    @ONLY
-    )
+# CONFIGURE_FILE(${PROJECT_SOURCE_DIR}/packaging/pkgconfig/openthreads.pc.in
+#         ${PROJECT_BINARY_DIR}/packaging/pkgconfig/openthreads.pc
+#     @ONLY
+#     )
 
-INSTALL(FILES ${PROJECT_BINARY_DIR}/packaging/pkgconfig/openthreads.pc DESTINATION lib${LIB_POSTFIX}/pkgconfig  COMPONENT libopenthreads-dev)
+#INSTALL(FILES ${PROJECT_BINARY_DIR}/packaging/pkg_config/openthreads.pc DESTINATION lib${LIB_POSTFIX}/pkgconfig  COMPONENT libopenthreads-dev)
