diff --git a/poppler/GlobalParamsWin.cc b/poppler/GlobalParamsWin.cc
index f989fb8..7ec1b8f 100644
--- a/poppler/GlobalParamsWin.cc
+++ b/poppler/GlobalParamsWin.cc
@@ -274,10 +280,15 @@ DisplayFontParam *GlobalParams::getDisplayFont(GfxFont *font) {
     DisplayFontParam *  dfp;
     GooString *         fontName = font->getName();
     char *              substFontName = NULL;
+    char                appDir[MAX_PATH];
 
     if (!fontName) return NULL;
     lockGlobalParams;
-    setupBaseFonts(NULL);
+    if (::GetModuleFileName(0, appDir, MAX_PATH) > 0)
+        setupBaseFonts(appendToPath(grabPath(appDir), "fonts")->getCString());
+    else
+        setupBaseFonts(NULL);
+    
     dfp = (DisplayFontParam *)displayFonts->lookup(fontName);
     if (!dfp) {
         substFontName = findSubstituteName(fontName->getCString());

