diff -bruN proj-4.9.1.original/src/Makefile.in proj-4.9.1/src/Makefile.in
--- proj-4.9.1.original/src/Makefile.in	2016-04-05 10:59:38.834624359 +0200
+++ proj-4.9.1/src/Makefile.in	2016-04-05 11:00:11.073813213 +0200
@@ -400,7 +400,7 @@
 multistresstest_LDADD = libproj.la -lpthread
 test228_LDADD = libproj.la -lpthread
 lib_LTLIBRARIES = libproj.la
-libproj_la_LDFLAGS = -no-undefined -version-info 9:0:0
+libproj_la_LDFLAGS = -no-undefined -version-info 0:0:0
 libproj_la_SOURCES = \
 	pj_list.h \
 	PJ_aeqd.c PJ_gnom.c PJ_laea.c PJ_mod_ster.c \
