From 59c9b6ace8c06c3aa8a74abf9e3e3b02b5687cd2 Mon Sep 17 00:00:00 2001
From: Mark Brand <mabrand@mabrand.nl>
Date: Fri, 2 Jun 2017 18:23:20 +0200
Subject: [PATCH] disable mapboxgl for MXE


diff --git a/src/plugins/geoservices/geoservices.pro b/src/plugins/geoservices/geoservices.pro
index 0810d392..de781ed1 100644
--- a/src/plugins/geoservices/geoservices.pro
+++ b/src/plugins/geoservices/geoservices.pro
@@ -10,7 +10,8 @@ qtConfig(c++14):!win32|mingw:!qnx {
     !exists(../../3rdparty/mapbox-gl-native/CMakeLists.txt) {
         warning("Submodule mapbox-gl-native does not exist. Run 'git submodule update --init' on qtlocation.")
     } else {
-        SUBDIRS += mapboxgl ../../3rdparty/mapbox-gl-native
-        mapboxgl.depends = ../../3rdparty/mapbox-gl-native
+#        SUBDIRS += mapboxgl ../../3rdparty/mapbox-gl-native
+#        mapboxgl.depends = ../../3rdparty/mapbox-gl-native
+	message(MXE: mapboxgl disabled)
     }
 }
-- 
2.11.0

