diff -bur qwt_qt4-5.2.orig/qwtconfig.pri qwt_qt4-5.2/qwtconfig.pri
--- qwt_qt4-5.2.orig/qwtconfig.pri	2014-10-08 18:19:07.388918558 +0530
+++ qwt_qt4-5.2/qwtconfig.pri	2014-10-08 18:20:25.676555425 +0530
@@ -38,7 +38,7 @@
 isEmpty(VVERSION) {
 
     # Qt 3
-    CONFIG           += debug     # release/debug
+    CONFIG           += release     # release/debug
 }
 else {
     # Qt 4
@@ -49,12 +49,12 @@
         # might need a debug version. 
         # Enable debug_and_release + build_all if you want to build both.
 
-        CONFIG           += debug     # release/debug/debug_and_release
+        CONFIG           += release     # release/debug/debug_and_release
         #CONFIG           += debug_and_release
         #CONFIG           += build_all
     }
     else {
-        CONFIG           += debug     # release/debug
+        CONFIG           += release     # release/debug
     }
 }
 
