--- a/src/gui_impl/commands/zoom_gui_command.h	2010-08-02 08:10:42.000000000 +0200
+++ b/src/gui_impl/commands/zoom_gui_command.h	2017-06-19 01:34:36.567643490 +0200
@@ -57,8 +57,8 @@
     //-------------------------------------------------------------------------
     float32 minPixelPerSample ();
 
-    static float const ZOOM_FACTOR_ = 2;
-    static float const MAX_HORIZONTAL_ZOOM_IN_ = 32;
+    const float ZOOM_FACTOR_ = 2.0;
+    const float MAX_HORIZONTAL_ZOOM_IN_ = 32.0;
 
     static QString const GOTO_;
     static QString const ZOOM_IN_VERTICAL_;
