diff --git a/src/stimfit/gui/app.cpp b/src/stimfit/gui/app.cpp
index b818e0c..f036a54 100755
--- a/src/stimfit/gui/app.cpp
+++ b/src/stimfit/gui/app.cpp
@@ -348,7 +348,11 @@ bool wxStfApp::OnInit(void)
 int wxStfApp::OnExit()
 {
 #if wxUSE_CONFIG
+#if (wxCHECK_VERSION(3, 0, 0))
+    GetDocManager()->FileHistorySave(*wxConfigBase::Get());
+#else
     GetDocManager()->FileHistorySave(*wxConfig::Get());
+#endif
 #endif // wxUSE_CONFIG
 
     delete GetDocManager();
