diff --git a/Makefile.static.in b/Makefile.static.in
index d7da6a5c..991f87b6 100644
--- a/Makefile.static.in
+++ b/Makefile.static.in
@@ -222,7 +222,9 @@ LIBS   += $(LIBLAPACK_LDFLAGS)
 CXXFLAGS += $(CFLAGS) $(CPPFLAGS)
 
 ## BIOSIG related stuff ##
-LIBS  += $(shell $(PKGCONF) --libs libbiosig)
+LIBS  += $(shell $(PKGCONF) --libs libbiosig) -ltinyxml
+LIBS  += $(shell $(PKGCONF) --libs openblas) -lgomp
+LIBS  += -lhdf5 -lhdf5_hl
 
 LIBS     += -lfftw3
 
