diff --git a/src/libstfio/intan/intanlib.cpp b/src/libstfio/intan/intanlib.cpp
index ed8dfa5..81eefdc 100644
--- a/src/libstfio/intan/intanlib.cpp
+++ b/src/libstfio/intan/intanlib.cpp
@@ -239,7 +239,13 @@ std::vector<std::vector<float> > read_aux_data(BinaryReader& binreader, uint16_t
 
 void stfio::importIntanFile(const std::string &fName, Recording &ReturnData, ProgressInfo& progDlg) {
     unique_ptr<FileInStream> fs(new FileInStream());
-	std::wstring wfName(fName.begin(), fName.end());
+
+#ifdef _MSC_VER
+    std::wstring wfName(fName.begin(), fName.end());
+#else
+    std::string wfName(fName);
+#endif
+
     fs->open(wfName);
 
     unique_ptr<BinaryReader> binreader(new BinaryReader(move(fs)));
diff --git a/src/libstfio/intan/intanlib.h b/src/libstfio/intan/intanlib.h
index fcef6e2..6b180c9 100644
--- a/src/libstfio/intan/intanlib.h
+++ b/src/libstfio/intan/intanlib.h
@@ -42,7 +42,7 @@ Example:
 
 #include "./../stfio.h"
 
-#ifdef _WINDOWS
+#ifdef _MSC_VER
 typedef __int8 int8_t;
 typedef __int16 int16_t;
 typedef __int32 int32_t;
diff --git a/src/stimfit/stf.h b/src/stimfit/stf.h
index ee15812..f44aeb8 100644
--- a/src/stimfit/stf.h
+++ b/src/stimfit/stf.h
@@ -29,6 +29,10 @@
     #include <boost/function.hpp>
 #endif
 #endif
+
+#if (__GNUC__ > 5)
+  #include <functional>
+#endif
 #include <vector>
 #include <map>
 #include <string>
