diff --git a/src/stimfit/gui/childframe.cpp b/src/stimfit/gui/childframe.cpp
index 0faf0518..adc3d126 100755
--- a/src/stimfit/gui/childframe.cpp
+++ b/src/stimfit/gui/childframe.cpp
@@ -173,7 +173,7 @@ void wxStfChildFrame::CreateMenuTraces(const std::size_t value) {
     pSize=new wxStaticText( m_traceCounter, wxID_ANY, wxEmptyString);
     wxString sizeStr;
 
-    pSpinCtrlTraceSizer->Add( pIndexText,     0, wxALIGN_CENTER_VERTICAL  | wxALL, 1) ;
+    pSpinCtrlTraceSizer->Add( pIndexText,     0, wxALL, 1) ;
     pSpinCtrlTraceSizer->Add( trace_spinctrl, 0, wxALIGN_LEFT, 1) ;
     pSpinCtrlTraceSizer->Add( pSize,          0, wxALIGN_LEFT  | wxALIGN_CENTER, 1) ;
 
