diff --git a/configure.ac b/configure.ac
index a464260d..0a7ddaf4 100644
--- a/configure.ac
+++ b/configure.ac
@@ -332,7 +332,7 @@ if test "${HDF5PREFIX}" = ""; then
 fi
 
 AC_CHECK_HEADER([hdf5.h], [], [AC_MSG_ERROR([Couldn't find hdf5 header])])
-AC_CHECK_LIB([hdf5],[H5Fopen],HAVE_HDF5="yes")
+AC_CHECK_LIB([hdf5],[H5Fopen],HAVE_HDF5="yes",,[-lz])
 if test "${HAVE_HDF5}" != "yes" ; then
     AC_MSG_ERROR([Couldn't find hdf5 libraries.])
 else
