--- subversion-1.9.2/configure.ac	2015-07-27 01:03:10.000000000 +0200
+++ subversion-1.9.2.new/configure.ac	2015-12-09 13:58:49.000000000 +0100
@@ -467,9 +467,6 @@
 
 dnl Mac OS specific features -------------------
 
-SVN_LIB_MACHO_ITERATE
-SVN_LIB_MACOS_PLIST
-SVN_LIB_MACOS_KEYCHAIN
 
 dnl APR_HAS_DSO -------------------
 
@@ -1271,7 +1268,6 @@
 # Python: Used for testsuite, and bindings
 
 
-PYTHON="`$abs_srcdir/build/find_python.sh`"
 if test -z "$PYTHON"; then
   AC_MSG_WARN([Python 2.7 or later is required to run the testsuite])
   AC_MSG_WARN([or to use the Subversion Python bindings])

--- subversion-1.9.2/subversion/libsvn_subr/config_win.c	2015-04-17 06:00:18.000000000 +0200
+++ subversion-1.9.2.new/subversion/libsvn_subr/config_win.c	2015-12-10 09:49:11.645120955 +0100
@@ -34,7 +34,7 @@
 #define WIN32_LEAN_AND_MEAN
 /* winsock2.h includes windows.h */
 #include <winsock2.h>
-#include <Ws2tcpip.h>
+#include <ws2tcpip.h>
 
 #include <shlobj.h>
 
--- subversion-1.9.2/subversion/libsvn_subr/win32_xlate.c	2015-02-03 13:56:57.000000000 +0100
+++ subversion-1.9.2.new/subversion/libsvn_subr/win32_xlate.c	2015-12-10 09:48:56.725192873 +0100
@@ -36,7 +36,7 @@
 
 /* winsock2.h includes windows.h */
 #include <winsock2.h>
-#include <Ws2tcpip.h>
+#include <ws2tcpip.h>
 #include <mlang.h>
 
 #include <apr.h>
@@ -53,6 +53,8 @@
 
 #include "svn_private_config.h"
 
+#define INITGUID 1
+DEFINE_GUID (IID_IMultiLanguage, 0x275c23e1,0x3747,0x11d0,0x9f,0xea,0x00,0xaa,0x00,0x3f,0x86,0x46);
 static svn_atomic_t com_initialized = 0;
 
 /* Initializes COM and keeps COM available until process exit.
@@ -139,10 +139,12 @@
       return saved; /* probably SVN_ERR_ATOMIC_INIT_FAILURE */
     }
 
+#if 0
   hr = CoCreateInstance(&CLSID_CMultiLanguage, NULL, CLSCTX_INPROC_SERVER,
                         &IID_IMultiLanguage, (void **) &mlang);
 
   if (FAILED(hr))
+#endif
     return APR_EGENERAL;
 
   /* Convert page name to wide string. */
--- subversion-1.9.2/subversion/libsvn_subr/io.c	2015-09-11 06:00:13.000000000 +0200
+++ subversion-1.9.2.new/subversion/libsvn_subr/io.c	2015-12-10 10:21:22.272463424 +0100
@@ -1789,7 +1789,7 @@
         }
     }
 
-    SVN_ERR(svn_utf__win32_utf8_to_utf16(&(const WCHAR*)buffer, source,
+    SVN_ERR(svn_utf__win32_utf8_to_utf16((const WCHAR**)&buffer, source,
                                          prefix, result_pool));
 
     /* Convert slashes to backslashes because the \\?\ path format
--- subversion-1.9.2/build/generator/gen_make.py	2015-07-27 01:03:10.000000000 +0200
+++ subversion-1.9.2.new/build/generator/gen_make.py	2015-12-09 13:58:49.000000000 +0100
@@ -633,7 +633,7 @@
         lib_required_private=[],
         )
       # libsvn_foo -> -lsvn_foo
-      data.lib_deps.append('-l%s' % lib_name.replace('lib', '', 1))
+      data.lib_deps.append('-l%s-1' % lib_name.replace('lib', '', 1))
       for lib_dep in lib_deps.split():
         if lib_dep == 'apriconv':
           # apriconv is part of apr-util, skip it
--- subversion-1.9.2/build/generator/templates/pkg-config.in.ezt	2016-02-11 15:48:52.204400231 +0100
+++ subversion-1.9.2.new/build/generator/templates/pkg-config.in.ezt	2014-05-05 17:04:00.000000000 +0200
@@ -1,7 +1,7 @@
 prefix=@prefix@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
-includedir=@includedir@
+includedir=@includedir@/subversion-1
 
 Name: [lib_name]
 Description: [lib_desc]
