--- SuiteSparse-3.7.0/CHOLMOD/Makefile	2011-12-08 19:56:52.000000000 +0100
+++ SuiteSparse-3.6.0/CHOLMOD/Makefile	2011-01-25 21:05:49.000000000 +0100
@@ -13,6 +13,6 @@
 
 # Compile the C-callable libraries and the Demo programs.
 all:
+	( cd Lib ; $(MAKE) )
-	( cd Demo ; $(MAKE) )
 
 # Compile the C-callable libraries only.
--- SuiteSparse-3.7.0/UMFPACK/Makefile	2011-12-08 20:57:27.000000000 +0100
+++ SuiteSparse-3.6.1/UMFPACK/Makefile	2011-01-25 21:07:50.000000000 +0100
@@ -13,5 +13,5 @@
 # compile all C code (except hb, fortran, and fortran64), including AMD and the
 # demos, but not the MATLAB mexFunctions
 all:
+	( cd Lib    ; $(MAKE) )
-	( cd Demo   ; $(MAKE) )
 
