--- a/src/common/utilscmn.cpp	2013-10-09 21:44:03.000000000 +0200
+++ b/src/common/utilscmn.cpp	2013-10-09 21:43:39.000000000 +0200
@@ -633,7 +633,11 @@ static bool ReadAll(wxInputStream *is, wxArrayString& output)
     // Notice that wxTextInputStream doesn't work correctly with wxConvAuto
     // currently, see #14720, so use the current locale conversion explicitly
     // under assumption that any external program should be using it too.
-    wxTextInputStream tis(*is, " \t", wxConvLibc);
+    wxTextInputStream tis(*is, " \t"
+#if wxUSE_UNICODE
+                                    , wxConvLibc
+#endif
+                                                );
 
     for ( ;; )
     { 
